aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -526,6 +529,200 @@ void cpu_hflip_vec(at::TensorIterator& iter) {
+    // flip(R) ->
+    // R = [ 31 (28 29 30) (25 26 27) (22 23 24) (19 20 21) (16 17 18) | 16 (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3) ]
+    //
+    // Write in 2 parts
+    // Output pointer:                                                                                                       v
+    // - Init:
+    //                (X X X)  (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X) (X X X) (X X X)
+    // 0) Move to initial position: output_ptr += (size0 - delta/2) * stride;
+    //                                                                                                                           v
+    //                (X X X)  (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X)    (X X X) (X X X) (X X X)
+    // - In the loop:
+    // 1) Move to write position: output_ptr -= usable_vec_stride and write 1st block from output_ptr + usable_vec_half_stride
+    //                     v                                                    | ----> vec_size / 2 ----------------------------|
+    // Output part 1: (X X X)  (X X X)    (X X X)    (X X X)    (X X X)    (X X 16)   (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3)
+    // 2) Write 2nd block from output_ptr:
+    //                     v ----> vec_size / 2 -----------------------------------|
+    // Output part 2: (X X 31) (28 29 30) (25 26 27) (22 23 24) (19 20 21) (16 17 18) (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3)
+    //
+    // - After the loop:
+    // 3) Move to write position: output_ptr -= (size0 - delta/2) * stride;
+    //                 v
+    //                (X X 31) (28 29 30) (25 26 27) (22 23 24) (19 20 21) (16 17 18) (13 14 15) (10 11 12) (7 8 9) (4 5 6) (1 2 3)
+
+    // Generate shuffle mask for N channels, N in [2, 8]
+    char mdata[16] = {};
+    for (int k=0; k<16; k++) {
+      int j = k / size0 + 1;
+      int v = (j * size0 - 1) - (k % size0);
+      v = std::min(v, (int) 15);
+      mdata[k] = v;
+    }
+    __m256i mask = _mm256_set_epi8(
+      mdata[0], mdata[1], mdata[2], mdata[3],
+      mdata[4], mdata[5], mdata[6], mdata[7],
+      mdata[8], mdata[9], mdata[10], mdata[11],
+      mdata[12], mdata[13], mdata[14], mdata[15], // first 128-bit lane
+
+      mdata[0], mdata[1], mdata[2], mdata[3],
+      mdata[4], mdata[5], mdata[6], mdata[7],
+      mdata[8], mdata[9], mdata[10], mdata[11],
+      mdata[12], mdata[13], mdata[14], mdata[15] // second 128-bit lane
+    );
+
+    const auto usable_vec_size = 2 * ((vec_size / 2) / size0);
+    const auto delta = vec_size - usable_vec_size * size0;
+    const auto usable_vec_stride = usable_vec_size * size0 * stride;
+    const auto usable_vec_half_stride = usable_vec_stride / 2;
+
+    __m256i data_vec, reversed_vec;
+
+    auto output_ptr = data[0];
+    auto input_ptr = data[1];
+
+    output_ptr += (size0 - delta/2) * stride;
```

Comment:
NIT: Why not just do 
```suggestion
    output_ptr += size0 - vec_size / 2;
```
Then `output_ptr` is in the correct place to write the first vector and all subsequent offsets are multiples of `usable_vec_half_stride`.